src/battle_util.c

```diff
@@ -32,8 +31,655 @@
 #include "constants/hold_effects.h"
 #include "constants/items.h"
 #include "constants/moves.h"
-#include "constants/species.h"
 #include "constants/weather.h"
+#include "battle_arena.h"
+#include "battle_pyramid.h"
+#include "international_string_util.h"
+#include "safari_zone.h"
+#include "sound.h"
+#include "task.h"
+#include "trig.h"
+#include "window.h"
+#include "constants/songs.h"
+
+extern const u8 *const gBattleScriptsForMoveEffects[];
+extern const u8 *const gBattlescriptsForBallThrow[];
+extern const u8 *const gBattlescriptsForRunningByItem[];
+extern const u8 *const gBattlescriptsForUsingItem[];
+extern const u8 *const gBattlescriptsForSafariActions[];
+
+static const u8 sPkblToEscapeFactor[][3] = {{0, 0, 0}, {3, 5, 0}, {2, 3, 0}, {1, 2, 0}, {1, 1, 0}};
+static const u8 sGoNearCounterToCatchFactor[] = {4, 3, 2, 1};
+static const u8 sGoNearCounterToEscapeFactor[] = {4, 4, 4, 4};
+
+void HandleAction_UseMove(void)
```

Comment:
Sync these functions based on the gist provided on Discord. The functions were modified before being moved to battle_util.